ui/src/track-layout/layout-map-api.ts

```diff
@@ -84,7 +115,7 @@ const locationTrackEndsCache = asyncCache<string, MapAlignmentEndPoints>();
 const referenceLineEndsCache = asyncCache<string, MapAlignmentEndPoints>();
 const sectionsWithoutProfileCache = asyncCache<string, AlignmentHighlight[]>();
 const sectionsWithoutLinkingCache = asyncCache<string, AlignmentHighlight[]>();
-const trackNumberTrackMeterCache = asyncCache<LayoutTrackNumberId, TrackMeter | undefined>();
+const trackNumberTrackMeterCache = asyncCache<string, TrackMeter | undefined>();
```

Comment:
Cacheavaimet taitaa olla vanhastaankin enimmäkseen stringejä koska ne koostuu usein monista arvoista. Noillekin olisi kiva olla tyypit mutta ts:ssä ei ole dataluokkia joten se menisi instanssi-vertailuksi defaulttina. Tuonne meidän cache-tyyppiin voisi koodata sen sisään käyttämällä vaikkapa json-serialisointia avaimiin... mutta se menee jo ihan toiseksi rempaksi.

Reply:
Käytännössähän cacheavaimien tarkemmalla tyypityksellä on hyvin vaikea voittaa paljoa mitään, kun näitä cacheja käytetään vain API-moduulien sisällä, yleensä vain yhdestä funktiosta, ja sielläkin yleensä erillinen cacheKey-funktio määrittää joka tapauksessa, mitä cacheavaimeen oikeasti menee. Lähinnä olen ottanut hittiä siitä, että väitetty tyyppi oli vaan väärin ja johti sillä harhaan.

Reply:
TypeScriptissähän sinällään pystyy määrittämään string-muotoisille tyypeille formaatin jollaisia niiden tulee olla (esimerkki `geometry-api.ts`:stä, jossa yhden cache-avaimen tyyppi on määritetty tarkemmin):

```
const locationTrackLinkingSummaryCache = asyncCache<
    `${LocationTrackId}_${PublicationState}_${LayoutDesignId}`,
    PlanLinkingSummaryItem[]
>();
```

Itse kyllä kannattaisin tuollaista tarkempaa tyypitystä, koska sitten voitaisiin olla varmempia että cache-avaimia ei voi määritellä väärin. Mutta ei mielestäni tämän tiketin asiaa, vaan tästä voisi tehdä omansa